controllers/storagecluster/cephtoolbox.go

```diff
@@ -117,21 +120,32 @@ func removeDuplicateTolerations(tolerations []corev1.Toleration) []corev1.Tolera
 	return list
 }
 
-func getMultusPublicNetwork(sc *ocsv1.StorageCluster) (string, error) {
+func (r *StorageClusterReconciler) getMultusPublicNetwork(sc *ocsv1.StorageCluster) (string, error) {
 	err := validateMultusSelectors(sc.Spec.Network.Selectors)
 	if err != nil {
 		return "", err
 	}
 
-	multusNetNs, err := util.GetOperatorNamespace()
-	if err != nil {
-		return "", err
+	if sc.Spec.Network.Selectors["public"] == "" {
+		return "", nil
 	}
 
-	if sc.Spec.Network.Selectors["public"] != "" {
-		multusNetName := sc.Spec.Network.Selectors["public"]
-		return fmt.Sprintf("%s/%s", multusNetNs, multusNetName), nil
+	multusNetName := sc.Spec.Network.Selectors["public"]
+	multusNetNamespacedName := strings.Split(multusNetName, "/")
+	nadNS, nadName := "", ""
+	if len(multusNetNamespacedName) == 1 {
+		nadNS, nadName = os.Getenv(util.OperatorNamespaceEnvVar), multusNetName
+	} else if len(multusNetNamespacedName) == 2 {
+		nadNS, nadName = multusNetNamespacedName[0], multusNetNamespacedName[1]
+	} else {
+		return "", fmt.Errorf("public network selector for multus: %s, is invalid", multusNetName)
```

Comment:
I think we should add the spec path in the logs so that customers would be aware of what exactly needs to be changed.